### PR TITLE
Just check name of exception to avoid an issue if ClassCastException …

### PR DESCRIPTION
**Pull request by baldersheim**
…comes from different class loader

@bjorncs or @jonmv PR

### CODE REVIEW

**Review by jonmv — COMMENTED** [review]
Are you sure this can happen?

**Comment by jonmv** [comment]
I guess this doesn't hurt, but I didn't think this was possible.

**Review by bratseth — APPROVED** [review]
I think this needs a comment or it might quickly revert back

**Review by bjorncs — COMMENTED** [review]
Sure about this?

**Comment by baldersheim** [comment]
No, this is just be my guess to how you can end up with
java.lang.IllegalArgumentException: java.lang.ClassCastException@1b0496ad
	at jdk.internal.reflect.GeneratedMethodAccessor304.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at com.yahoo.processing.request.CloneHelper.cloneByReflection(CloneHelper.java:121)
	at com.yahoo.processing.request.CloneHelper.objectClone(CloneHelper.java:99)
	at com.yahoo.search.query.properties.CloneHelper.objectClone(CloneHelper.java:17)

**Comment by bjorncs** [comment]
We need the full stack trace to be sure. My guess is that whatever code path logging the exception only includes the cause (not outer exception), or some code of the stack is wrapping or partially unwrapping the exception before rethrowing.

File: container-core/src/main/java/com/yahoo/processing/request/CloneHelper.java [hunk 1]
```diff
@@ -98,7 +98,7 @@ else if (object instanceof HashSet)
         try {
             return cloneByReflection(object);
         } catch (IllegalArgumentException e) {
-            if ( ! (e.getCause() instanceof ClassCastException))
+            if ( ! (e.getCause() != null && ClassCastException.class.getName().equals(e.getCause().getClass().getName())))
```
Comment (bjorncs) [inline container-core/src/main/java/com/yahoo/processing/request/CloneHelper.java:101]: Log exception before `if` with level `FINE` and enable verbose logging for this class through feature flag.

**Review by bjorncs — CHANGES_REQUESTED** [review]

**Comment by jonmv** [comment]
We have strong confirmation the cause is indeed `null`, and we're confused by a message string which is `ClassCastException@<some hash>` ... Ü

**Pull request closed without merging**